### PR TITLE
Feature/5082 high risk caused by multiple low risks

### PR DESCRIPTION
**Pull request by naveeddotio**
## Description
Display information in contact journal if the high risk is cause by multiple low risks
The condition in the tech specs says that we have to display this message when there is high risk and minimumDistinctEncountersWithHighRisk is zero.

See here:
https://github.com/corona-warn-app/cwa-app-tech-spec/compare/release/1.14.x..feature/contact-journal-multiple-low-risk-encounters

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5082

## Screenshots
Simulator Screen Shot - Clone 1 of iPhone 11 - 2021-02-23 at 13 30 29
Simulator Screen Shot - Clone 1 of iPhone 11 - 2021-02-23 at 13 32 43

### CODE REVIEW

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings [hunk 1]
```diff
@@ -425,9 +425,9 @@ Bei ausgeschalteter Hintergrundaktualisierung müssen Sie die App täglich aufru
 
 "OnboardingInfo_alwaysStayInformedPage_title" = "Warnungen erhalten, Risiken erkennen";
 
-"OnboardingInfo_alwaysStayInformedPage_boldText" = "Die App kann Sie automatisch über Ihren Risikostatus informieren und bei Neuinfektionen von Menschen, denen Sie begegnet sind, warnen. Erlauben Sie der App jetzt, Sie zu benachrichtigen.";
+"OnboardingInfo_alwaysStayInformedPage_boldText" = "Die App kann Sie automatisch über Ihren Risikostatus informieren und bei Neuinfektionen von Menschen, denen Sie begegnet sind, warnen. Erlauben Sie der App jetzt, Sie zu benachrichtigen.";
 
-"OnboardingInfo_alwaysStayInformedPage_normalText" = "Auf diese Weise können Sie sich zum Schutz Ihrer Mitmenschen in Quarantäne begeben und sich nach entsprechender Abklärung testen lassen.";
+"OnboardingInfo_alwaysStayInformedPage_normalText" = "Auf diese Weise können Sie sich zum Schutz Ihrer Mitmenschen in Quarantäne begeben und sich nach entsprechender Abklärung testen lassen.";
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings:430]: What are these changes?

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/Cells/DiaryOverviewDayCellModel.swift [hunk 2]
```diff
@@ -77,11 +79,13 @@ final class DiaryOverviewDayCellModel {
...
-			return selectedEntries.isEmpty ?
-				AppStrings.ContactDiary.Overview.riskTextStandardCause :
-				[AppStrings.ContactDiary.Overview.riskTextStandardCause, AppStrings.ContactDiary.Overview.riskTextDisclaimer].joined(separator: "\n")
-
+		case let .encounter(risk):
+			switch risk {
+			case .low:
+				return selectedEntries.isEmpty ? AppStrings.ContactDiary.Overview.riskTextStandardCause : [AppStrings.ContactDiary.Overview.riskTextStandardCause, AppStrings.ContactDiary.Overview.riskTextDisclaimer].joined(separator: "\n")
+			case .high:
+				return computeExposureHistoryDetail()
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/Cells/DiaryOverviewDayCellModel.swift:87]: You could move all this logic here. I don't really like the naming `compute...`

```suggestion
			case .high where minimumDistinctEncountersWithHighRisk > 0:
				return diaryDay.selectedEntries.isEmpty ? AppStrings.ContactDiary.Overview.riskTextStandardCause : [AppStrings.ContactDiary.Overview.riskTextStandardCause, AppStrings.ContactDiary.Overview.riskTextDisclaimer].joined(separator: "\n")
							case .high where minimumDistinctEncountersWithHighRisk >= 0:
				return diaryDay.selectedEntries.isEmpty ? AppStrings.ContactDiary.Overview.riskTextLowRiskEncountersCause : [AppStrings.ContactDiary.Overview.riskTextLowRiskEncountersCause, AppStrings.ContactDiary.Overview.riskTextDisclaimer].joined(separator: "\n")
```

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/DiaryOverviewViewModel.swift [hunk 3]
```diff
@@ -91,4 +93,10 @@ class DiaryOverviewViewModel {
 		return .encounter(riskLevelPerDate)
 	}
 
+	private func getMinimumDistinctEncountersWithHighRisk(by date: Date) -> Int {
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/DiaryOverviewViewModel.swift:96]:
```suggestion
	private func minimumDistinctEncountersWithHighRisk(by date: Date) -> Int {
```

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/Cells/DiaryOverviewDayCellModel.swift [hunk 4]
```diff
@@ -94,9 +98,17 @@ final class DiaryOverviewDayCellModel {
 	var formattedDate: String {
 		diaryDay.formattedDate
 	}
+	
+	func computeExposureHistoryDetail() -> String {
+		if minimumDistinctEncountersWithHighRisk > 0 {
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/Cells/DiaryOverviewDayCellModel.swift:103]: `> 0` is problematic if you return -1 for non-existent days. I guess it's better to check for `== 0` and return the `.riskTextLowRiskEncountersCause` then. I changed it already in the suggestion above.

**Review by nickguendling — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/Cells/DiaryOverviewDayCellModel.swift (hunk 4 above)
Comment (naveeddotio) [reply src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/Cells/DiaryOverviewDayCellModel.swift:103]: This is on purpose i.e., -1 for non existent days. 

- UI element _High Risk due to Low Risk Encounters_ - displayed if `Risk Level by Date` for the date associated with the day of the list item is `High` and `Minimum Distinct Encounters With High Risk per Date` for that day is `0` or does not have a value. (-1 in our case)

Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/Cells/DiaryOverviewDayCellModel.swift:103]: Okay, that seems to be correctly implemented according to the tech spec then 👌🏼

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/DiaryOverviewViewModel.swift (hunk 3 above)
Comment (naveeddotio) [reply src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/DiaryOverviewViewModel.swift]: Fixed.

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/Cells/DiaryOverviewDayCellModel.swift (hunk 2 above)
Comment (naveeddotio) [reply src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/Cells/DiaryOverviewDayCellModel.swift]: Fixed

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings (hunk 1 above)
Comment (naveeddotio) [reply src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings]: Could be due to merge, reverted them.

**Review by nickguendling — APPROVED** [review]

**Review by 30mar — APPROVED** [review]

**Pull request merged**